src/main/java/com/excelsiorjet/api/tasks/ClasspathEntry.java

```diff
@@ -18,37 +18,154 @@
+        } else {
+            return null;
+        }
+    }
+
+    private static OptimizationType deriveOptimize(DependencySettings dependencySettings) {
+        if (dependencySettings.optimize == null && dependencySettings.isLibrary != null) {
+            return dependencySettings.isLibrary ? OptimizationType.AUTO_DETECT : OptimizationType.ALL;
+        } else if (dependencySettings.optimize != null) {
+            return OptimizationType.fromString(dependencySettings.optimize);
+        } else {
+            return null;
+        }
+    }
+
+    public ClasspathEntry(File path, ProtectionType protect, OptimizationType optimize, PackType pack, String packagePath, Boolean disableCopyToPackage, boolean isMainArtifact) {
+        this.path = path;
+        this.protect = protect;
+        this.optimize = optimize;
+        this.pack = pack;
+        if (packagePath != null) {
+            this.packagePath = packagePath;
+        } else {
+            this.packagePath = isMainArtifact ? "" : null;
+        }
+        this.disableCopyToPackage = disableCopyToPackage;
+        this.isMainArtifact = isMainArtifact;
+    }
 
-    public ClasspathEntry(File file, boolean isLib) {
-        this.file = file;
-        this.isLib = isLib;
+    public ClasspathEntry isMainArtifact(boolean isMainArtifact) {
+        return new ClasspathEntry(path, protect, optimize, pack, packagePath, disableCopyToPackage, isMainArtifact);
     }
 
     /**
-     * File that is pointed by classpath entry.
+     * Classpath entry optimization type.
      */
-    public File getFile() {
-        return file;
+    public enum OptimizationType {
+
+        ALL("all", "all"),
+        AUTO_DETECT("auto-detect", "autodetect");
+
+        public final String userValue;
+        public final String jetValue;
+
+        OptimizationType(String userValue, String jetValue) {
+            this.userValue = userValue;
+            this.jetValue = jetValue;
+        }
+
+        public static OptimizationType fromString(String optimizationType) {
+            try {
+                return OptimizationType.valueOf(Utils.parameterToEnumConstantName(optimizationType));
+            } catch (Exception e) {
+                return null;
+            }
+        }
+
     }
 
     /**
-     * Flag indicating is classpath entry points to part of application (that should be protected and optimized, if requested)
-     * or not.
+     * Classpath entry pack type.
      */
-    public boolean isLib() {
-        return isLib;
+    public enum PackType {
+
+        NONE("none", "none"),
+        AUTO_DETECT("auto-detect", "noncompiled"),
+        ALL("all", "all");
+
+        public final String userValue;
+        public final String jetValue;
+
+        PackType(String userValue, String jetValue) {
+            this.userValue = userValue;
+            this.jetValue = jetValue;
+        }
+
+        public static PackType fromString(String packType) {
+            try {
```

Comment:
probably it is better to handle null value explicitly as it is permitted situation here